src/averageAges.js

```diff
@@ -15,11 +15,19 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const CENTURY_DIVIDER = 100;
+  const filteredPeople = people.filter(person =>
+    (century !== undefined
+    && Math.ceil(person.died / CENTURY_DIVIDER) === century
+    && person.sex === 'm')
+    || (century === undefined && person.sex === 'm')
+  );
+
+  const totalAge = filteredPeople.reduce((sum, person) =>
+    sum + (person.died - person.born), 0);
```

Comment:
Avoid using 'reduce' directly on 'filteredPeople' array without checking if it's empty. This could throw an exception if the array is empty. A possible fix could be to add a condition check before 'reduce'.